data_model/src/block.rs

```diff
@@ -189,10 +205,10 @@ impl SignedBlock {
     /// Add additional signatures to this block
     #[must_use]
     #[cfg(feature = "transparent_api")]
-    pub fn sign(mut self, key_pair: &KeyPair) -> Self {
+    pub fn sign(mut self, private_key: &PrivateKey, node_pos: u64) -> Self {
         let SignedBlock::V1(block) = &mut self;
-        let signature = iroha_crypto::SignatureOf::new(key_pair, &block.payload);
-        block.signatures.insert(signature);
+        let signature = SignatureOf::new(private_key, &block.payload);
+        block.signatures.push(BlockSignature(node_pos, signature));
```

Comment:
Implication here is that now it's possible to store multiple signatures under the same `node_pos`.
Can't this be used by malicious peer to add irrelevant signatures to block?